include/nbl/system/CArchiveLoaderZip.h

```diff
@@ -0,0 +1,297 @@
+		memset(&entry.header, 0, sizeof(SZIPFileHeader));
+
+		SGZIPMemberHeader header;
+		system::future<size_t> headerFuture;
+		m_file->read(headerFuture, &header, read_offset, sizeof(SGZIPMemberHeader));
+		read_offset += sizeof(SGZIPMemberHeader);
+		if (headerFuture.get() == sizeof(SGZIPMemberHeader))
+		{
+			// check header value
+			if (header.sig != 0x8b1f)
+				return false;
+			// now get the file info
+			if (header.flags & EGZF_EXTRA_FIELDS)
+			{
+				// read lenth of extra data
+				uint16_t dataLen;
+
+				system::future<size_t> lenFuture;
+				m_file->read(lenFuture, &dataLen, read_offset, 2);
+				read_offset += 2;
+			}
+			std::filesystem::path zipFileName = "";
+			if (header.flags & EGZF_FILE_NAME)
+			{
+				char c;
+				{
+					system::future<size_t> fut;
+					m_file->read(fut, &c, read_offset++, 1);
+				}
+				while (c)
+				{
+					zipFileName += c;
+					system::future<size_t> fut;
+					m_file->read(fut, &c, read_offset++, 1);
+				}
+			}
+			else
+			{
+				// no file name?
+				zipFileName = m_file->getFileName().filename();
+			
+				// rename tgz to tar or remove gz extension
+				if (zipFileName.extension() == ".tgz")
+				{
+					zipFileName.string()[zipFileName.string().size() - 2] = 'a';
+					zipFileName.string()[zipFileName.string().size() - 1] = 'r';
+				}
+				else if (zipFileName.extension() == ".gz")
+				{
+					zipFileName.string()[zipFileName.string().size() - 3] = 0;
+				}
+			}
+			if (header.flags & EGZF_COMMENT)
+			{
+				char c = 'a';
+				while (c)
+				{
+					system::future<size_t> fut;
+					m_file->read(fut, &c, read_offset++, 1);
+				}
+			}
+			if (header.flags & EGZF_CRC16)
+				read_offset += 2;
+
+			entry.Offset = read_offset;
+			entry.header.FilenameLength = zipFileName.native().length();
+			entry.header.CompressionMethod = header.compressionMethod;
+			entry.header.DataDescriptor.CompressedSize = (m_file->getSize() - 8) - read_offset;
+
+			read_offset += entry.header.DataDescriptor.CompressedSize;
+
+			// read CRC
+			{
+				system::future<size_t> fut;
+				m_file->read(fut, &entry.header.DataDescriptor.CRC32, read_offset, 4);
+				read_offset += 4;
+			}
+			// read uncompressed size
+			{
+				system::future<size_t> fut;
+				m_file->read(fut, &entry.header.DataDescriptor.UncompressedSize, read_offset, 4);
+				read_offset += 4;
+			}
+
+			addItem(ZipFileName, entry.Offset, entry.header.DataDescriptor.UncompressedSize, false, 0);
+			FileInfo.push_back(entry);
+		}
+		return false;
+	}
+	bool scanZipHeader()
+	{
+
+	}
+};
+
+class CArchiveLoaderZip : public IArchiveLoader
+{
+public:
+	virtual bool isALoadableFileFormat(IFile* file) const override
+	{
+		std::string ext = system::extension_wo_dot(file->getFileName().extension());
+		auto available_exts = getAssociatedFileExtensions();
+		const char* current_ext = available_exts[0];
+		while (current_ext)
+		{
+			if (ext == current_ext)
+			{
+				return true;
+			}
+			++current_ext;
+		}
+		return false;
```

Comment:
nope, thats not how checking if something is loadable should work

when we check if something if loadable, we always dismiss the filename and extension, we look at the actual contents of the file (usually headers)

Reply:
fixed